src/lib/transport.js

```diff
@@ -37,17 +40,24 @@ function createPacketAndCondition ({
   const address = destinationAccount + '.~' + protocol + '.' + receiverId +
     (id ? ('.' + id) : '')
 
-  const blobData = omitUndefined({
-    expires_at: expiresAt,
-    data: data
-  })
+  const details = createDetails({
+    unsafeHeaders: Object.assign({
+      'Key-Id': receiverId, // TODO: is this right?
```

Comment:
From the article you linked it looks like `Content-HMAC` contains the digest of the data, rather than information on an HMAC key.

Also, did we decide that the second part of `hmac-sha-256 W2WuqXqCedW3hj4UsQdzBg` is going to be used as a nonce? It sounds like we're keeping the receiverId in the ILP address.